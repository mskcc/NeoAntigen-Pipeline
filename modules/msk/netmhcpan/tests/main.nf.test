nextflow_process {

    name "Test Process NETMHCPAN"
    script "../main.nf"
    process "NETMHCPAN"

    tag "modules"
    tag "modules_nfcore"
    tag "netmhcpan"

    test("test_neoantigen") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_short_maf'], checkIfExists: true),
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['winners_hla_txt'], checkIfExists: true),
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.fastaSequences,
                    file(process.out.xls[0][1]).name,
                    file(process.out.xls[0][2]).name,
                    file(process.out.netmhcpanoutput[0][1]).name,
                    file(process.out.netmhcpanoutput[0][2]).name
                    ).match()
                }
            )
        }

    }

}
