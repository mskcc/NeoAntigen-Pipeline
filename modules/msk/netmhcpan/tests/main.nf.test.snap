{
    "test_neoantigen": {
        "content": [
            [
                "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.mutated_sequences.fa:md5,a998392759482f8a1c63c795cfd9e399",
                    "test.WT_sequences.fa:md5,27e677ffa032a87934df4c62e9f15357"
                ]
            ],
            "test.MUT.xls",
            "test.WT.xls",
            "test.MUT.netmhcpan.output",
            "test.WT.netmhcpan.output"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-17T22:17:33.67688"
    }
}